hearthstone/simulator/core/hero.py

```diff
@@ -95,15 +107,70 @@ def on_buy_step(self):
     def battlecry_multiplier(self) -> int:
         return 1
 
-    def select_discover(self, discover_index: 'DiscoverIndex'):
+    def select_discover(self, discover_index: 'DiscoverIndex', context: 'BuyPhaseContext'):
         pass
 
     def valid_select_discover(self, discover_index: 'DiscoverIndex') -> bool:
-        return bool(self.discover_choices) and discover_index in range(len(self.discover_choices))
+        return bool(self.discover_queue) and discover_index in range(len(self.discover_queue[0]))
 
     def hero_info(self) -> Optional[str]:
         return None
 
+    def handle_secrets(self, event: 'CardEvent', context: Union['BuyPhaseContext', 'CombatPhaseContext']):
```

Comment:
It might make sense to do the same treatment as you did to spells for secrets: Make a player.secrets spell, make each secret a class, and just have a handle_event() function on it.